apps/src/code-studio/components/AdvancedShareOptions.jsx

```diff
@@ -154,20 +159,48 @@ class AdvancedShareOptions extends React.Component {
   publishExpoExport = async () => {
     this.setState({exportingExpo: 'publish'});
     try {
-      const expoUri = await this.props.exportApp({mode: 'expoPublish'});
+      const {expoUri, expoSnackId} = await this.props.exportApp({
+        mode: 'expoPublish'
+      });
       this.setState({
         exportingExpo: null,
-        expoUri
+        exportExpoError: null,
+        expoUri,
+        expoSnackId
       });
     } catch (e) {
       this.setState({
         exportingExpo: null,
+        expoUri: null,
+        expoSnackId: null,
         exportExpoError:
           'Failed to publish project to Expo. Please try again later.'
       });
     }
   };
 
+  generateExpoApk = async () => {
+    const {expoSnackId} = this.state;
+    this.setState({generatingExpoApk: true});
+    try {
+      const expoApkUri = await this.props.exportApp({
+        mode: 'expoGenerateApk',
+        expoSnackId
+      });
+      this.setState({
+        generatingExpoApk: false,
+        generatingExpoApkError: null,
+        expoApkUri
+      });
+    } catch (e) {
+      this.setState({
+        generatingExpoApk: false,
+        generatingExpoApkError:
+          'Failed to create Android app. Please try again later.'
+      });
```

Comment:
Would it be helpful to add some monitoring here and track what sort of errors we see?

Reply:
yes, good idea. Although while under experiment, it's just a few of us who will be doing this.